### PR TITLE
relayed v3 features

### PR DESCRIPTION
**Pull request by cfaur09**
## Reasoning
-  RelayedV3 implementation is now live on testnet, starting from release T1.7.14.0.
  
## Proposed Changes
- extracting the all informations regarding `innerTransactions`
- adding new filter to extract all the transfers where the relayer is `erd1...`

## How to test
- `<api>/transfers?relayer=erd1ua84zk6tkz92k8l5n8fp97yws0ztzkmcyu234tltn873lsr6q8xqgg52j3` -> should return all transfers filtered by relayer
- `<api>/transfers?size=10000` -> some transfers should contain innerTransactions field
- `<api>/transfers` -> some transfers should contains `relayer` field

### CODE REVIEW

File: src/common/gateway/entities/transaction.inner.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  nonce: number = 0;
+  value: string = '';
+  receiver: string = '';
+  sender: string = '';
+  gasPrice: number = 0;
+  gasLimit: number = 0;
+  data: string = '';
+  signature: string = '';
+  chainId: number = 0;
```
Comment (andreibancioiu) [inline src/common/gateway/entities/transaction.inner.ts:14]: Perhaps `chainID` to be consistent with the Proxy / Gateway?

https://devnet-gateway.multiversx.com/transaction/c01cae88fb9c9fd6f7a82406fd63c36f186953d3436dcce13d7651ed89a4c893?withResults=true

Additionally, it should be a string (not a number).

File: src/common/gateway/entities/transaction.inner.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  version: number = 0;
```
Comment (andreibancioiu) [inline src/common/gateway/entities/transaction.inner.ts:15]: Can also add:
 -  `options` (as a number)
 - `guardianSignature` (as a string)
 - `senderUsername`, `receiverUsername` (as strings)

File: src/endpoints/transactions/entities/transaction.detailed.ts [hunk 3]
```diff
@@ -64,5 +65,9 @@ export class TransactionDetailed extends Transaction {
   @Field(() => String, { description: "Relayed transaction version.", nullable: true })
   @ApiProperty({ type: String, nullable: true })
   relayedVersion: string | undefined = undefined;
+
+  @Field(() => [TransactionInner], { description: 'Inner transactions list details.', nullable: true })
+  @ApiProperty({ type: TransactionInner, isArray: true })
+  innerTransactions: TransactionDetailed[] | undefined = undefined;
```
Comment (andreibancioiu) [inline src/endpoints/transactions/entities/transaction.detailed.ts:71]: `TransactionInner` or `TransactionDetailed`?

File: src/common/gateway/entities/transaction.inner.ts [hunk 4]
```diff
@@ -11,7 +11,10 @@ export class TransactionInner {
   gasLimit: number = 0;
   data: string = '';
   signature: string = '';
-  chainId: number = 0;
+  chainID: string = '';
   version: number = 0;
   relayer: string = '';
+  options: number = 0;
+  guardianSignature: string = '';
+  senderUsername: string = '';
```
Comment (andreibancioiu) [inline src/common/gateway/entities/transaction.inner.ts:19]: Either drop it, or keep it, but also add `receiverUsername`?

**Review by andreibancioiu — DISMISSED** [review]

File: src/endpoints/transactions/transaction.service.ts [hunk 5]
```diff
@@ -152,7 +152,11 @@ export class TransactionService {
     const elasticTransactions = await this.indexerService.getTransactions(filter, pagination, address);
 
     let transactions: TransactionDetailed[] = [];
-    transactions = elasticTransactions.map(x => ApiUtils.mergeObjects(new TransactionDetailed(), x));
+         transactions = elasticTransactions.map(x => {
+           const transaction = ApiUtils.mergeObjects(new TransactionDetailed(), x);
+           transaction.relayedVersion = x.version;
```
Comment (andreibancioiu) [inline src/endpoints/transactions/transaction.service.ts:157]: ` relayedVersion` and `transaction version` are different concepts, just to double check if everything is all right.

Comment (cfaur09) [reply src/endpoints/transactions/transaction.service.ts]: reverted

Comment (cfaur09) [reply src/endpoints/transactions/transaction.service.ts]: done

File: src/common/gateway/entities/transaction.inner.ts (hunk 4 above)
Comment (cfaur09) [reply src/common/gateway/entities/transaction.inner.ts:19]: receiverUsername added

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**